### PR TITLE
Change test settings

### LINKED ISSUE #94277: [CI] SearchableSnapshotsIntegTests testMaxRestoreBytesPerSecIsUsed failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/rrbr7cc3ka7wa/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests/testMaxRestoreBytesPerSecIsUsed

**Reproduction line:**
```
./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testMaxRestoreBytesPerSecIsUsed" -Dtests.seed=1DF63FBE262BEEE -Dtests.locale=pt-BR -Dtests.timezone=America/Yakutat -Druntime.java=17
```

**Applicable branches:**
8.6

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests&tests.test=testMaxRestoreBytesPerSecIsUsed

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a value greater than <0L>
     but: <0L> was equal to <0L>

  at __randomizedtesting.SeedInfo.seed([1DF63FBE262BEEE:2C8B91E9D0FD49BC]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testMaxRestoreBytesPerSecIsUsed(SearchableSnapshotsIntegTests.java:510)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by volodk85**
Over past year there were 5 observed [failures](https://ci-stats.elastic.co/s/jenkins/app/discover#/?_g=h@00a189c&_a=h@01ee4fa) if the test. The test asserts non-zero cumulative rate limiting delay, simply speaking it checks if rate limiting is happening.

Rate limiting logic involves two components:`RateLimitingInputStream` (ES codebase)  and `RateLimiter` (Lucene codebase) where cumulative delay is collected behind `listener.onPause(pause)` in line 47 and then asserted in the test.
https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/server/src/main/java/org/elasticsearch/index/snapshots/blobstore/RateLimitingInputStream.java#L44-L48

Rate limiting counter can remain eq to 0 **iff** there were no invocation of `listener.onPause`, meaning `pause` was eq 0.
The value is calculated on a fly by `RateLimiter`, if `startNS >= targetNS` condition does not hold, it returns 0. 
```
public long pause(long bytes) {

      long startNS = System.nanoTime();

      double secondsToPause = (bytes / 1024. / 1024.) / mbPerSec;

      long targetNS;

      // Sync'd to read + write lastNS:
      synchronized (this) {

        // Time we should sleep until; this is purely instantaneous
        // rate (just adds seconds onto the last time we had paused to);
        // maybe we should also offer decayed recent history one?
        targetNS = lastNS + (long) (1000000000 * secondsToPause);

        if (startNS >= targetNS) {
          // OK, current time is already beyond the target sleep time,
          // no pausing to do.

          // Set to startNS, not targetNS, to enforce the instant rate, not
          // the "averaaged over all history" rate:
          lastNS = startNS;
          return 0;
        }
```
Problem with this code is that `startNS = System.nanoTime()` is executed **after** the physical read from the stream happened. The only linking piece between (1) physical IO read and (2) `startNS` variable  is `secondsToPause` which tries to *estimate* the delay. If the arbitrary delay happens between (1) and (2) code returns ZERO. The failure is easy to reproduce by setting a breaking point on `startNS = System.nanoTime()` and passing it (thus simulating delay) - the test always fails.

To reduce likelihood of the failure I'm offering to tweak two available parameters in `secondsToPause = (bytes / 1024. / 1024.) / mbPerSec` formula - `bytes` and `mbPerSec`. By increasing the ratio, I expect to hold the `startNS >= targetNS` condition more often.

In my mind, the test is OK'ish itself by there is lack of control over RateLimiter component which prevents making the test better.

If we keep seeing failures, than we need to re-work way of asserting fact of rate limiting.   

Fix #94277

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java [hunk 1]
```diff
@@ -463,7 +463,9 @@ public void testMaxRestoreBytesPerSecIsUsed() throws Exception {
             // we compute the min across all the max shard sizes by node in order to
             // trigger the rate limiter in all nodes. We could just use the min shard size
             // but that would make this test too slow.
-            long rateLimitInBytes = getMaxShardSizeByNodeInBytes(indexName).values().stream().min(Long::compareTo).get();
+            // Update: setting rate limit at a 90% of minimax value above
```
Comment (fcofdez) [inline x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:466]: We should add a small explanation about why we're using the 90% of the minmax value to set the rate limiter.
Also, we rely on the version control system to track updates.

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java [hunk 2]
```diff
@@ -451,7 +451,7 @@ public void testMaxRestoreBytesPerSecIsUsed() throws Exception {
             true,
             false,
             IntStream.range(0, nbDocs)
-                .mapToObj(i -> client().prepareIndex(indexName).setSource("foo", randomBoolean() ? "bar" : "baz"))
+                .mapToObj(i -> client().prepareIndex(indexName).setSource("foo", randomAlphaOfLength(1048)))
```
Comment (fcofdez) [inline x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:454]: does this change make the test more stable?

**Review by fcofdez — COMMENTED** [review]
Looks good, I left a couple of comments.

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java (hunk 2 above)
Comment (volodk85) [reply x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:454]: As I tried to explain in initial PR description, I'd like to increase value of `secondsToPause = (bytes / 1024. / 1024.) / mbPerSec` ratio.

Document size (highlighted lines above) eventually translated to `bytes` param in this formula, whereas 90% of minimax goes to `mbPerSec` parameter. By increasing numerator and decreasing denominator we are getting larger value for `secondsToPause`. all these could tigger rate limiting logic more often and hence make test more stable.

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java (hunk 1 above)
Comment (volodk85) [reply x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:466]: I think it is already easy to track back to this PR and this writing by looking on commit history. Would you like to include link on this PR to the comments?

Comment (fcofdez) [reply x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java]: I usually prefer to have all the context in the code, just a small comment would be great 👍

**Review by fcofdez — APPROVED** [review]
LGTM

**Pull request merged**